### PR TITLE
Cql generation drool

### PR DESCRIPTION
**Pull request by jreyno77**
**Description**

- Github Issue:  
- [ ] I've read the contribution guidelines 
- [ ] Code compiles without errors
- [ ] Tests are created / updated
- [ ] Documentation is created / updated

By creating this PR you acknowledge that your contribution will be licensed under Apache 2.0

### CODE REVIEW

**Comment by JPercival** [comment]
All the code needs to be moved to the `org.opencds.cqf.tooling` package. `individual_tooling` is not a valid top-level package for this artifact.

File: pom.xml [hunk 1]
```diff
@@ -73,55 +85,161 @@
...
+            <artifactId>mts-support-rckms</artifactId>
+            <version>2.0.0</version>
+        </dependency>
+
         <!-- WikiMedia to HTML conversion library (-QdmToQiCore) -->
         <dependency>
             <groupId>info.bliki.wiki</groupId>
             <artifactId>bliki-core</artifactId>
-            <version>3.1.0</version>
+            <version>3.1.1-SNAPSHOT</version>
```
Comment (JPercival) [inline pom.xml:242]: We should only use SNAPSHOT versions when absolutely necessary. They block a release to Maven.

File: pom.xml [hunk 2]
```diff
@@ -296,14 +474,6 @@
...
                         <release>8</release>
-                        <failOnWarning>true</failOnWarning>
-                        <showWarnings>true</showWarnings>
-                        <compilerArgs>
-                            <arg>-Xlint:all</arg>
-                            <!-- This fixes a boostrapclass path warning which the "release" property fixes on Java 9+ -->
-                            <arg>-Xlint:-options</arg>
-                            <arg>-Werror</arg>
-                        </compilerArgs>
                     </configuration>
```
Comment (JPercival) [inline pom.xml:477]: This should not be disabled. The code should be updated to not have warnings, or if there are specific issues that are impossible to work around they should be suppressed individually.

As it stands the current code won't run on a Java 8 VM due to the fact some Java 9 APIS are being used. (e.g. "List.of", "Map.of")

File: pom.xml [hunk 3]
```diff
@@ -319,7 +489,7 @@
                         </archive>
                     </configuration>
                 </plugin>
-                <plugin>
+                <!-- <plugin>
```
Comment (JPercival) [inline pom.xml:492]: Lack of Javadoc is also a blocker for release to Maven.

File: src/main/java/org/opencds/cqf/tooling/Main.java [hunk 4]
```diff
@@ -163,11 +163,20 @@
...
+
+        - VmrToFhirTransformer
+            - command: mvn exec: java -Dexec.args="-VmrToFhir -ifp=./src/test/resources/org/opencds/cqf/tooling/operation/VmrToFhir -op=./src/test/resources/org/opencds/cqf/tooling/operation/VmrToFhir/vMROutput.xml -e=xml"
+            - this tooling transforms vMR data to FHIR data
         */
 
 public class Main {
 
     public static void main(String[] args) {
+        System.console().readLine();
```
Comment (JPercival) [inline src/main/java/org/opencds/cqf/tooling/Main.java:179]: This needs to be removed.

File: src/main/java/org/opencds/cqf/tooling/cql_generation/builder/VmrToModelElmBuilder.java [hunk 5]
```diff
@@ -0,0 +1,889 @@
...
+    public final ObjectFactory of = new ObjectFactory();
+    public final org.hl7.cql_annotations.r1.ObjectFactory af = new org.hl7.cql_annotations.r1.ObjectFactory();
+    protected String modelIdentifier;
+    protected String modelVersion;
+    protected String modelUri;
+    protected LibrarySourceProvider lsp;
+    protected IncludeDef includeHelper;
+    protected DecimalFormat decimalFormat;
+    protected Logger logger;
+    protected Map<String, Marker> markers = Map.of(
```
Comment (JPercival) [inline src/main/java/org/opencds/cqf/tooling/cql_generation/builder/VmrToModelElmBuilder.java:81]: Several of instances things like this that won't run on Java 8.

File: src/main/java/org/opencds/cqf/tooling/cql_generation/drool/adapter/DefinitionAdapter.java [hunk 6]
```diff
@@ -0,0 +1,234 @@
+package org.opencds.cqf.tooling.cql_generation.drool.adapter;
```
Comment (JPercival) [inline src/main/java/org/opencds/cqf/tooling/cql_generation/drool/adapter/DefinitionAdapter.java:1]: Generally, an "adapter" is used to connect one interface to another. In this case a whole new object is being created, so I'd suggest using "builder" or "converter" instead. Same applies throughout.

File: src/main/java/org/opencds/cqf/tooling/utilities/ResourceUtils.java [hunk 7]
```diff
@@ -211,10 +211,14 @@ public static FhirContext getFhirContext(FhirVersion fhirVersion) {
     private static Map<String, IBaseResource> getR4DepLibraryResources(String path, Map<String, IBaseResource> dependencyLibraries, FhirContext fhirContext, Encoding encoding, Boolean versioned) {      
       List<String> dependencyLibraryPaths = getR4DepLibraryPaths(path, fhirContext, encoding, versioned);
       for (String dependencyLibraryPath : dependencyLibraryPaths) {
-        Object resource = IOUtils.readResource(dependencyLibraryPath, fhirContext);
-        if (resource instanceof org.hl7.fhir.r4.model.Library) {
-          org.hl7.fhir.r4.model.Library library = (org.hl7.fhir.r4.model.Library)resource;
-          dependencyLibraries.putIfAbsent(library.getId(), library);
+        if (dependencyLibraryPath.contains("ModelInfo")) {
+          System.out.println("skip");
```
Comment (JPercival) [inline src/main/java/org/opencds/cqf/tooling/utilities/ResourceUtils.java:215]: Should use logger.debug or logger.trace for this. It's generally considered bad practice to white debug output to stdout.

File: src/test/resources/org/opencds/cqf/individual_tooling/cql_generation/drool/visitor/Abdominal_Cramps_314c4.xml [hunk 8]
```diff
@@ -0,0 +1,137 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JPercival) [inline src/test/resources/org/opencds/cqf/individual_tooling/cql_generation/drool/visitor/Abdominal_Cramps_314c4.xml:1]: This file (and others) are still under the "individual_tooling" package. Needs to move to "tooling"

**Review by JPercival — CHANGES_REQUESTED** [review]

**Review by brynrhodes — CHANGES_REQUESTED** [review]
The PR shouldn't change the version to 1.3.1, should keep it at 1.3.1-SNAPSHOT. Also, this PR is changing quite a few other dependencies, can we verify those are absolutely necessary changes? It's not clear why the HAPI structure dependencies would change, for example.

**Review by JPercival — APPROVED** [review]

**Review by brynrhodes — APPROVED** [review]

**Pull request merged**